exporter/redis.go

```diff
@@ -436,16 +436,31 @@ func (e *Exporter) scrape(scrapes chan<- scrapeResult) {
 		defer c.Close()
 		log.Debugf("connected to: %s", addr)
 
+		is_cluster := false
+
 		info, err := redis.String(c.Do("INFO", "ALL"))
 		if err == nil {
 			err = e.extractInfoMetrics(info, addr, scrapes)
-		}
-		if err != nil {
+			if strings.Index(info, "cluster_enabled:1") != -1 {
+				is_cluster = true
+			}
+		} else {
 			log.Printf("redis err: %s", err)
 			errorCount++
 			continue
 		}
 
+		if is_cluster {
+			info, err = redis.String(c.Do("CLUSTER", "INFO"))
+			if err == nil {
+				err = e.extractInfoMetrics(info, addr, scrapes)
```

Comment:
Yeah, see #33 which Brian submitted to improve metrics names in line with Prometheus best practices.